frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -904,6 +903,10 @@ type dispatchState struct {
 	// workdirSet is set to true if a workdir has been set
 	// within the current dockerfile.
 	workdirSet bool
+
```

Comment:
possible follow-up: `workdirSet` to `healthcheck` seem like fields with similar usage. I think they could be all combined into one struct that is included here.